pkg/resourcemanager/mysql/server/reconcile.go

```diff
@@ -41,27 +52,42 @@ func (m *MySQLServerClient) Ensure(ctx context.Context, obj runtime.Object, opts
 		return true, err
 	}
 
-	createmode := mysql.CreateModeDefault
+	createMode := mysql.CreateModeDefault
 	if len(instance.Spec.CreateMode) != 0 {
-		createmode = mysql.CreateMode(instance.Spec.CreateMode)
+		createMode = mysql.CreateMode(instance.Spec.CreateMode)
 	}
 
 	// If a replica is requested, ensure that source server is specified
-	if createmode == mysql.CreateModeReplica {
+	if createMode == mysql.CreateModeReplica {
 		if len(instance.Spec.ReplicaProperties.SourceServerId) == 0 {
 			instance.Status.Message = "Replica requested but source server unspecified"
 			return true, nil
 		}
 	}
 
+	adminCreds, err := m.GetUserProvidedAdminCredentials(ctx, instance)
```

Comment:
Should we make this resource reconcile when the secret is updated? Are people going to expect the user details to be updated in that case? (Is updating the credentials like that supported in ARM?)

Reply:
We probably should, yes. It's a bit of feature creep because we didn't support rolling credentials before (since the user couldn't specify them), but really we should now.

I'll look into it but may do it as a separate PR as it'll have another set of tests ensuring that the update actually works, and the actual watching of that secret may add some complexity.

Reply:
I've filed #1634 - will follow up with a separate PR tackling that.